toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/FeatureForm.kt

```diff
@@ -106,6 +118,119 @@ public sealed class FeatureFormEditingEvent {
+     *
+     * @since 300.0.0
+     */
+    public data class GroupResult(
+        val element: UtilityAssociationsFormElement,
+        val utilityAssociationGroupResult: UtilityAssociationGroupResult
+    ) : FeatureFormNavigationRoute()
+
+    /**
+     * Indicates the route for [UtilityAssociationResult] details screen.
+     *
+     * @param element The [UtilityAssociationsFormElement] associated with the association result.
+     * @param utilityAssociationResult The selected [UtilityAssociationResult].
+     *
+     * @since 300.0.0
+     */
+    public data class AssociationResult(
+        val element: UtilityAssociationsFormElement,
+        val utilityAssociationResult: UtilityAssociationResult
+    ) : FeatureFormNavigationRoute()
+
+    /**
+     * Indicates the route for selecting a [UtilityAssociationFeatureSource] when adding a new
+     * association.
+     *
+     * @param element The [UtilityAssociationsFormElement] where the association is being added.
+     *
+     * @since 300.0.0
+     */
+    public data class SelectAssociationFeatureSource(
+        val element: UtilityAssociationsFormElement
+    ) : FeatureFormNavigationRoute()
+
+    /**
+     * Indicates the route for selecting a [UtilityAssetType] from a [UtilityAssociationFeatureSource]
+     * when adding a new association.
+     *
+     * @param element The [UtilityAssociationsFormElement] where the association is being added.
+     * @param featureSource The selected [UtilityAssociationFeatureSource].
+     *
+     * @since 300.0.0
+     */
+    public data class SelectUtilityAssetType(
+        val element: UtilityAssociationsFormElement,
+        val featureSource: UtilityAssociationFeatureSource
+    ) : FeatureFormNavigationRoute()
+
+    /**
+     * Indicates the route for selecting a [UtilityAssociationFeatureCandidate] from a
+     * [UtilityAssetType] when adding a new association.
+     *
+     * @param element The [UtilityAssociationsFormElement] where the association is being added.
+     * @param featureSource The selected [UtilityAssociationFeatureSource] the asset type belongs to.
+     * @param assetType The selected [UtilityAssetType].
+     *
+     * @since 300.0.0
+     */
+    public data class SelectAssociationFeatureCandidate(
+        val element: UtilityAssociationsFormElement,
+        val featureSource: UtilityAssociationFeatureSource,
+        val assetType: UtilityAssetType
+    ) : FeatureFormNavigationRoute()
+
+    /**
+     * Indicates the route for creating a new association with the selected
+     * [UtilityAssociationFeatureCandidate].
+     *
+     * @param element The [UtilityAssociationsFormElement] where the association is being added.
+     * @param featureSource The selected [UtilityAssociationFeatureSource] the candidate belongs to.
+     * @param candidate The selected [UtilityAssociationFeatureCandidate].
+     *
+     * @since 300.0.0
+     */
+    public data class CreateAssociation(
```

Comment:
```suggestion
    public data class UtilityAssociationCreateAssociation(
```

Also missing `UtilityAssociationsFilter`